### PR TITLE
fix: block not declared routes, tests: add single decrlared route tests

### PR DESCRIPTION
**Pull request by RGO230**

### CODE REVIEW

File: src/MediaRouter.php [hunk 1]
```diff
@@ -23,12 +23,17 @@ public function media()
...
+            $options = array_combine(array_values($options), array_values($options));
+
             if (empty($options)){
                 $options = array_fill_keys(array_keys($defaultOptions), true);
             } else {
-                $options = array_merge($defaultOptions, array_fill_keys(array_keys($defaultOptions), true));
+                $options = array_merge($defaultOptions, array_fill_keys(array_keys($options), true));
             }
 
+
```
Comment (DenTray) [inline src/MediaRouter.php:36]: please revert this change

File: src/MediaRouter.php [hunk 2]
```diff
@@ -23,12 +23,17 @@ public function media()
                 'search' => false,
             ];
 
+            $options = array_column($options, 'value');
+
+            $options = array_combine(array_values($options), array_values($options));
```
Comment (DenTray) [inline src/MediaRouter.php:28]: looks strange for me, maybe we need to call array_fill_keys here? 

please also mo this logic into the `else` block

**Review by DenTray — CHANGES_REQUESTED** [review]

File: src/MediaRouter.php [hunk 3]
```diff
@@ -26,7 +26,9 @@ public function media()
             if (empty($options)){
                 $options = array_fill_keys(array_keys($defaultOptions), true);
             } else {
-                $options = array_merge($defaultOptions, array_fill_keys(array_keys($defaultOptions), true));
+                $options = array_column($options, 'value');
+                $options = array_fill_keys(array_values($options), array_keys($options));
+                $options = array_merge($defaultOptions, array_fill_keys(array_keys($options), true));
             }
```
Comment (DenTray) [inline src/MediaRouter.php:32]:
```php
foreach ($options as $option) {
    $defaultOptions[$option->value] = true;
}
```

**Review by DenTray — CHANGES_REQUESTED** [review]

File: src/MediaRouter.php [hunk 4]
```diff
@@ -23,17 +22,18 @@ public function media()
...
-                when($options['bulk_create'], fn () => $this->post('media/bulk', 'bulkCreate')->name('media.create.bulk'));
-                when($options['search'], fn () => $this->get('media', 'search')->name('media.search'));
+            $this->controller(MediaController::class)->group(function () use ($defaultOptions) {
+                when($defaultOptions['create'], fn() => $this->post('media', 'create')->name('media.create'));
+                when($defaultOptions['delete'], fn() => $this->delete('media/{id}', 'delete')->name('media.delete'));
+                when(
+                    $defaultOptions['bulk_create'],
+                    fn() => $this->post('media/bulk', 'bulkCreate')->name('media.create.bulk')
+                );
+                when($defaultOptions['search'], fn() => $this->get('media', 'search')->name('media.search'));
```
Comment (DenTray) [inline src/MediaRouter.php:36]:
```suggestion
                when($defaultOptions['create'], fn () => $this->post('media', 'create')->name('media.create'));
                when($defaultOptions['delete'], fn () => $this->delete('media/{id}', 'delete')->name('media.delete'));
                when($defaultOptions['bulk_create'], fn () => $this->post('media/bulk', 'bulkCreate')->name('media.create.bulk'));
                when($defaultOptions['search'], fn () => $this->get('media', 'search')->name('media.search'));
```

**Review by DenTray — APPROVED** [review]

**Review by DenTray — APPROVED** [review]

**Comment by RGO230** [comment]
tests failed by https://github.com/RonasIT/laravel-media/pull/30

**Review by DenTray — APPROVED** [review]

**Pull request merged**